### PR TITLE
fix: updated the text as per the ux diagram

### PR DESCRIPTION
**Pull request by debjyoti-pandit**
followed: https://marvelapp.com/prototype/2e221bb6/screen/85673714

-----------------------------------------------------------------------------------------------------------------

image

image

image

image

image

image

image

image

image

image

image

----------------------------------------------------------------------------------------------------------------------

image

image

image

-----------------------------------------------------------------------------------------------------------------------

image

image

image

image

image

### CODE REVIEW

**Review by alfonsomthd — APPROVED** [review]

**Review by avanthakkar — APPROVED** [review]

**Comment by SanjalKatiyar** [comment]
@debjyoti-pandit add screenshots of all the pages which are updated...

File: locales/en/plugin__mcg-ms-console.json [hunk 1]
```diff
@@ -2,30 +2,29 @@
...
   "Multi data source": "Multi data source",
-  "The bucket will serve reads from several selected data sources, creating a virtual namespace on top of them and will write to one or more of those as its chosen write target": "The bucket will serve reads from several selected data sources, creating a virtual namespace on top of them and will write to one or more of those as its chosen write target",
+  "The bucket reads from several selected data sources, creating a virtual namespace on top of them, and writes into one of those.": "The bucket reads from several selected data sources, creating a virtual namespace on top of them, and writes into one of those.",
   "Namespace": "Namespace",
-  "To allow access to this bucket, an ObjectBucketClaim will be created in the below specified namespace in the cluster. Select the namespace in which it will be created": "To allow access to this bucket, an ObjectBucketClaim will be created in the below specified namespace in the cluster. Select the namespace in which it will be created",
+  "Creates object bucket claim in the specified namespace.": "Creates object bucket claim in the specified namespace.",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:9]:
```suggestion
  "Creates ObjectBucketClaim in the specified namespace.": "Creates object bucket claim in the specified namespace.",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 2]
```diff
@@ -2,30 +2,29 @@
...
+  "Optionally, select the bucket you want to replicate data to.": "Optionally, select the bucket you want to replicate data to.",
   "Create": "Create",
   "Cancel": "Cancel",
-  "Buckets": "Buckets",
-  "Create new bucket": "Create new bucket",
-  "Create new Bucket": "Create new Bucket",
-  "A set of policies that would apply to all buckets (OBCs) created with the specific bucket policy. These policies include namespace and caching": "A set of policies that would apply to all buckets (OBCs) created with the specific bucket policy. These policies include namespace and caching",
+  "Create bucket policy": "Create bucket policy",
+  "Create new bucket policy": "Create new bucket policy",
+  "A set of policies which would apply to an object bucket claim created with the specific bucket policy. These policies include placement, namespace and caching.": "A set of policies which would apply to an object bucket claim created with the specific bucket policy. These policies include placement, namespace and caching.",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:16]:
```suggestion
  "A set of policies which would apply to an ObjectBucketClaim created with the specific bucket policy. These policies include placement, namespace and caching.": "A set of policies which would apply to an object bucket claim created with the specific bucket policy. These policies include placement, namespace and caching.",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 3]
```diff
@@ -78,60 +78,70 @@
   "Access Key Field": "Access Key Field",
   "Switch to Secret": "Switch to Secret",
   "Secret Key Field": "Secret Key Field",
-  "ObjectBucketClaim Name": "ObjectBucketClaim Name",
-  "my-object-bucket": "my-object-bucket",
+  "Object bucket claim name": "Object bucket claim name",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:81]:
```suggestion
  "ObjectBucketClaim name": "Object bucket claim name",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 4]
```diff
@@ -78,60 +78,70 @@
...
+  "my-object-bucket-claim": "my-object-bucket-claim",
   "If not provided a generic name will be generated.": "If not provided a generic name will be generated.",
-  "StorageClass": "StorageClass",
+  "Storage class": "Storage class",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:84]:
```suggestion
  "StorageClass": "Storage class",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 5]
```diff
@@ -78,60 +78,70 @@
...
   "Defines the object-store service and the bucket provisioner.": "Defines the object-store service and the bucket provisioner.",
-  "Bucket Policy": "Bucket Policy",
-  "Create ObjectBucketClaim": "Create ObjectBucketClaim",
+  "Bucket policy": "Bucket policy",
+  "Create new object bucket claim": "Create new object bucket claim",
+  "Create object bucket claim": "Create object bucket claim",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:88]:
```suggestion
  "Create new ObjectBucketClaim": "Create new object bucket claim",
  "Create ObjectBucketClaim": "Create object bucket claim",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 6]
```diff
@@ -78,60 +78,70 @@
...
+  "StorageClass": "StorageClass",
   "ObjectBuckets": "ObjectBuckets",
-  "Object Bucket Details": "Object Bucket Details",
-  "Object Bucket Claim": "Object Bucket Claim",
-  "ObjectBucket details": "ObjectBucket details",
+  "Object buckets": "Object buckets",
+  "Object bucket details": "Object bucket details",
+  "Object bucket claim": "Object bucket claim",
+  "Delete object bucket": "Delete object bucket",
+  "ObjectBucketClaims": "ObjectBucketClaims",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:96]:
```suggestion
  "ObjectBuckets": "Object buckets",
  "ObjectBucket details": "Object bucket details",
  "ObjectBucketClaim": "Object bucket claim",
  "Delete ObjectBucket": "Delete object bucket",
  "ObjectBucketClaims": "ObjectBucketClaims",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 7]
```diff
@@ -78,60 +78,70 @@
...
-  "Object Bucket Claim Details": "Object Bucket Claim Details",
-  "Object Bucket": "Object Bucket",
-  "ObjectBucketClaim details": "ObjectBucketClaim details",
-  "Object Bucket Claim Data": "Object Bucket Claim Data",
+  "Delete object bucket claim": "Delete object bucket claim",
+  "Create new": "Create new",
+  "Object bucket claim details": "Object bucket claim details",
+  "Object bucket": "Object bucket",
+  "Attach to deployment": "Attach to deployment",
+  "Object bucket claim data": "Object bucket claim data",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:103]:
```suggestion
  "Delete ObjectBucketClaim": "Delete object bucket claim",
  "Create new": "Create new",
  "ObjectBucketClaim details": "Object bucket claim details",
  "ObjectBucket": "Object bucket",
  "Attach to a Deployment": "Attach to deployment",
  "ObjectBucketClaim data": "Object bucket claim data",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 8]
```diff
@@ -78,60 +78,70 @@
...
-  "Connected Buckets": "Connected Buckets",
-  "Create data source": "Create data source",
+  "{{bucketsCount}} bucket policies": "{{bucketsCount}} bucket policies",
+  "Connected bucket policies": "Connected bucket policies",
+  "{{bucketsCount}} bucket policy": "{{bucketsCount}} bucket policy",
   "Target Bucket": "Target Bucket",
   "Target Blob Container": "Target Blob Container",
   "View more": "View more",
-  "Connected ObjectBucketClaims": "Connected ObjectBucketClaims",
+  "Connected object bucket claims": "Connected object bucket claims",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:144]:
```suggestion
  "Connected ObjectBucketClaims": "Connected object bucket claims",
```

File: packages/components/bucket-policy/create-bucket-policy.tsx [hunk 9]
```diff
@@ -135,11 +135,7 @@ const CreateBucketPolicy: React.FC<CreateBucketPolicyProps> = ({
       path: '/mcgms/cluster',
     },
     {
-      name: t('Buckets'),
-      path: '/mcgms/cluster/resource/noobaa.io~v1alpha1~BucketClass',
-    },
```
Comment (SanjalKatiyar) [inline packages/components/bucket-policy/create-bucket-policy.tsx:140]: why have we removed entire breadcrumb ?

File: packages/components/data-resource/create-data-resource.tsx [hunk 10]
```diff
@@ -28,11 +28,7 @@ const CreateDataResource: React.FC<CreateDataResourceProps> = ({
       path: '/mcgms/cluster',
     },
     {
-      name: t('Data source'),
-      path: '/mcgms/cluster/resource/noobaa.io~v1alpha1~NamespaceStore',
-    },
```
Comment (SanjalKatiyar) [inline packages/components/data-resource/create-data-resource.tsx:33]: same, why have we removed this ?

File: packages/components/mcg/ObjectBucketClaim.tsx [hunk 11]
```diff
@@ -262,9 +265,13 @@ export const OBCListPage: React.FC<PageProps> = (props) => {
...
       <Modal {...modalProps} />
-      <ListPageHeader title={t('ObjectBucketClaims')}>
+      <ListPageHeader
+        title={t('{{obcDisplayText}}', {
+          obcDisplayText: OBJECT_BUCKET_CLAIMS,
+        })}
+      >
         <ListPageCreateLink to={createLink}>
-          {t('Create ObjectBucketClaim')}
+          {t('Create new')}
```
Comment (SanjalKatiyar) [inline packages/components/mcg/ObjectBucketClaim.tsx:274]:
```suggestion
          {t('Create ObjectBucketClaim')}
          OR,
          {t('Create new ObjectBucketClaim')}
```
based upon how we show at other places.

File: locales/en/plugin__mcg-ms-console.json (hunk 1 above)
Comment (debjyoti-pandit) [reply locales/en/plugin__mcg-ms-console.json:9]: why are we doing this? any particular reason?

File: packages/components/bucket-policy/create-bucket-policy.tsx (hunk 9 above)
Comment (debjyoti-pandit) [reply packages/components/bucket-policy/create-bucket-policy.tsx:140]: it is part of the UX where we don't have the intermediatory list page bread crumb

https://marvelapp.com/prototype/2e221bb6/screen/86409130

File: packages/components/data-resource/create-data-resource.tsx (hunk 10 above)
Comment (debjyoti-pandit) [reply packages/components/data-resource/create-data-resource.tsx:33]: same as above and maintain uniformity

File: packages/components/bucket-policy/create-bucket-policy.tsx (hunk 9 above)
Comment (SanjalKatiyar) [reply packages/components/bucket-policy/create-bucket-policy.tsx:140]: we are coming to create page from the list page... so it make sense to show that in breadcrumb as well...

Comment (SanjalKatiyar) [reply packages/components/bucket-policy/create-bucket-policy.tsx:140]: let's keep it and not remove it !

File: locales/en/plugin__mcg-ms-console.json [hunk 12]
```diff
@@ -78,60 +79,68 @@
   "Access Key Field": "Access Key Field",
   "Switch to Secret": "Switch to Secret",
   "Secret Key Field": "Secret Key Field",
-  "ObjectBucketClaim Name": "ObjectBucketClaim Name",
-  "my-object-bucket": "my-object-bucket",
+  "ObjectBucketClaim name": "Object bucket claim name",
+  "my-object-bucket-claim": "my-object-bucket-claim",
   "If not provided a generic name will be generated.": "If not provided a generic name will be generated.",
-  "StorageClass": "StorageClass",
+  "StorageClass": "Storage class",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:85]:
```suggestion
  "StorageClass": "StorageClass",
```

File: locales/en/plugin__mcg-ms-console.json [hunk 13]
```diff
@@ -78,60 +79,68 @@
...
   "Defines the object-store service and the bucket provisioner.": "Defines the object-store service and the bucket provisioner.",
-  "Bucket Policy": "Bucket Policy",
-  "Create ObjectBucketClaim": "Create ObjectBucketClaim",
+  "Bucket policy": "Bucket policy",
+  "Create new ObjectBucketClaim": "Create new object bucket claim",
+  "Create ObjectBucketClaim": "Create object bucket claim",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:89]: keys and values can never be different... there is some issue... try `yarn i18n`

File: locales/en/plugin__mcg-ms-console.json [hunk 14]
```diff
@@ -78,60 +79,68 @@
...
   "Name": "Name",
-  "ObjectBuckets": "ObjectBuckets",
-  "Object Bucket Details": "Object Bucket Details",
-  "Object Bucket Claim": "Object Bucket Claim",
-  "ObjectBucket details": "ObjectBucket details",
+  "ObjectBuckets": "Object buckets",
+  "ObjectBucket details": "Object bucket details",
+  "ObjectBucketClaim": "Object bucket claim",
+  "Delete object bucket": "Delete object bucket",
+  "ObjectBucketClaims": "Object bucket claims",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:95]: same... something is wrong here... key and values should be same...

File: locales/en/plugin__mcg-ms-console.json [hunk 15]
```diff
@@ -78,60 +79,68 @@
...
   "Attach to Deployment": "Attach to Deployment",
-  "Object Bucket Claim Details": "Object Bucket Claim Details",
-  "Object Bucket": "Object Bucket",
-  "ObjectBucketClaim details": "ObjectBucketClaim details",
-  "Object Bucket Claim Data": "Object Bucket Claim Data",
+  "Delete ObjectBucketClaim": "Delete object bucket claim",
+  "ObjectBucketClaim details": "Object bucket claim details",
+  "Object bucket": "Object bucket",
+  "Attach to deployment": "Attach to deployment",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:100]:
```suggestion
  "Attach to Deployment": "Attach to deployment",
```

**Comment by debjyoti-pandit** [comment]
updated the description after the latest push

File: locales/en/plugin__mcg-ms-console.json [hunk 16]
```diff
@@ -78,60 +79,67 @@
...
+  "Bucket policy": "Bucket policy",
+  "Create new ObjectBucketClaim": "Create new ObjectBucketClaim",
   "Create ObjectBucketClaim": "Create ObjectBucketClaim",
   "Name": "Name",
   "ObjectBuckets": "ObjectBuckets",
-  "Object Bucket Details": "Object Bucket Details",
-  "Object Bucket Claim": "Object Bucket Claim",
   "ObjectBucket details": "ObjectBucket details",
+  "ObjectBucketClaim": "ObjectBucketClaim",
+  "Delete object bucket": "Delete object bucket",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:94]:
```suggestion
  "Delete object bucket": "Delete ObjectBucket",
```

File: cypress/tests/bucket-policy.spec.ts [hunk 17]
```diff
@@ -73,14 +73,14 @@ describe('Bucket policy page', () => {
...
-        .should('contain', '1 data source');
+        .should('contain', '1 resources');
       cy.log('Verify name of the connected data source');
       cy.byTestID('mcg-resource-popover').should('be.visible').click();
       cy.contains(DATA_SOURCE_NAME_NSFS);
       cy.log('Verify if OBC is created or not');
       cy.byTestID('obc-resource-popover')
         .should('be.visible')
-        .should('contain', '1 ObjectBucketClaim');
+        .should('contain', '1 Object bucket claims');
```
Comment (SanjalKatiyar) [inline cypress/tests/bucket-policy.spec.ts:83]:
```suggestion
        .should('contain', '1 ObjectBucketClaims');
```

File: locales/en/plugin__mcg-ms-console.json [hunk 18]
```diff
@@ -78,60 +79,67 @@
...
+  "ObjectBucketClaims": "ObjectBucketClaims",
   "Attach to Deployment": "Attach to Deployment",
-  "Object Bucket Claim Details": "Object Bucket Claim Details",
-  "Object Bucket": "Object Bucket",
+  "Delete ObjectBucketClaim": "Delete ObjectBucketClaim",
   "ObjectBucketClaim details": "ObjectBucketClaim details",
-  "Object Bucket Claim Data": "Object Bucket Claim Data",
+  "Object bucket": "Object bucket",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__mcg-ms-console.json:99]:
```suggestion
  "ObjectBucket": "Object bucket",
```

File: packages/components/data-resource/s3-endpoint-type.tsx [hunk 19]
```diff
@@ -334,6 +334,7 @@ export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
           id="target-bucket"
           value={state.target}
           data-test={`${type.toLowerCase()}-target-bucket`}
+          placeholder="Name of the target bucket"
```
Comment (SanjalKatiyar) [inline packages/components/data-resource/s3-endpoint-type.tsx:337]:
```suggestion
          placeholder=t("Name of the target bucket")
```

File: packages/components/resource-pages/bucket-policy-list-view.tsx [hunk 20]
```diff
@@ -211,10 +216,10 @@ export const BucketPolicyListView: React.FC = () => {
       resourceModel={NooBaaBucketClassModel}
       resourceMap={objectBucketClaimMap}
       tableColumns={tableColumns as TableColumn<K8sResourceCommon>[]}
-      createButtonTitle={t('Create bucket')}
+      createButtonTitle={t('Create new')}
```
Comment (SanjalKatiyar) [inline packages/components/resource-pages/bucket-policy-list-view.tsx:219]:
```suggestion
      createButtonTitle={t('Create new bucket policy')}
```
similarly for data source "Create new data source"

**Comment by SanjalKatiyar** [comment]
looks good, just 2 things:
1) In create button, we don't use "new". So for example: it will be "Create data source" not "Create new data source", similarly at all places.
2) Use: https://github.com/red-hat-storage/odf-console/blob/3f183a12d878136f5f240e1c7879b73ec3cb44b0/packages/odf/components/odf-dashboard/status-card/status-card.tsx#L153 to pluralize strings. I can at places we are mentioning for example "1 OBCs" etc.

**Comment by debjyoti-pandit** [comment]
image

image

image

----------------------------------------------------------------------------------------------------------------------------------------------------------------

image

image

----------------------------------------------------------------------------------------------------------------------------------------------------------------

image

image

----------------------------------------------------------------------------------------------------------------------------------------------------------------

image

image

image

----------------------------------------------------------------------------------------------------------------------------------------------------------------

**Pull request merged**